distributed/scheduler.py

```diff
@@ -5091,19 +5091,24 @@ async def restart(self, client=None, timeout=30):
                 stimulus_id=stimulus_id,
             )
 
-        nannies = {addr: ws.nanny for addr, ws in self.workers.items()}
+        nanny_workers: dict[str, str] = {
```

Comment:
> this annotation will become redundant once we overhaul annotations in the SchedulerState class. I'd rather put the annotations in the class declaration (C++-style)

I think this is worth opening an issue to discuss, can you take care of this, please? 
I had the impression that this kind of typing was mostly triggered by the Cythonization. Before we ask people to add or remove certain type annotations, we should settle how we want this to be done overall

Reply:
Yes, the explicit typing in the implementation code was 99% because of Cython. Annotating the instance attributes is sufficient in most cases. I'll open a ticket in community.